test/cmd/login.sh

```diff
@@ -0,0 +1,88 @@
+# setting --server bypasses the not configured message
+os::cmd::expect_failure_and_text "oc get services --server=http://${API_HOST}:${unused_port}" 'did you specify the right host or port'
+
+# Set KUBERNETES_MASTER for oc from now on
+export KUBERNETES_MASTER="${API_SCHEME}://${API_HOST}:${API_PORT}"
+
+# Set certificates for oc from now on
+if [[ "${API_SCHEME}" == "https" ]]; then
+    # test bad certificate
+    os::cmd::expect_failure_and_text "oc get services" 'certificate signed by unknown authority'
+fi
+
+# login and logout tests
+# bad token should error
+os::cmd::expect_failure_and_text "oc login ${KUBERNETES_MASTER} --certificate-authority='${MASTER_CONFIG_DIR}/ca.crt' --token=badvalue" 'The token provided is invalid or expired'
+# bad --api-version should error
+os::cmd::expect_failure_and_text "oc login ${KUBERNETES_MASTER} -u test-user -p test-password --api-version=foo/bar/baz" 'error.*foo/bar/baz'
+# --token and --username are mutually exclusive
+os::cmd::expect_failure_and_text "oc login ${KUBERNETES_MASTER} -u test-user --token=tmp --insecure-skip-tls-verify" 'mutually exclusive'
+# must only accept one arg (server)
+os::cmd::expect_failure_and_text "oc login https://server1 https://server2.com" 'Only the server URL may be specified'
+# logs in with a valid certificate authority
+os::cmd::expect_success "oc login ${KUBERNETES_MASTER} --certificate-authority='${MASTER_CONFIG_DIR}/ca.crt' -u test-user -p anything --api-version=v1"
+os::cmd::expect_success_and_text "cat ${HOME}/.kube/config" "v1"
+os::cmd::expect_success 'oc logout'
+# logs in skipping certificate check
+os::cmd::expect_success "oc login ${KUBERNETES_MASTER} --insecure-skip-tls-verify -u test-user -p anything"
+# logs in by an existing and valid token
+temp_token=$(oc config view -o template --template='{{range .users}}{{ index .user.token }}{{end}}')
+os::cmd::expect_success_and_text "oc login --token=${temp_token}" 'using the token provided'
+os::cmd::expect_success 'oc logout'
+# properly parse server port
+os::cmd::expect_failure_and_text 'oc login https://server1:844333' 'Not a valid port'
+# properly handle trailing slash
+os::cmd::expect_success "oc login --server=${KUBERNETES_MASTER} --certificate-authority='${MASTER_CONFIG_DIR}/ca.crt' -u test-user -p anything"
+# create a new project
+os::cmd::expect_success "oc new-project project-foo --display-name='my project' --description='boring project description'"
+os::cmd::expect_success_and_text "oc project" 'Using project "project-foo"'
+# new user should get default context
+os::cmd::expect_success "oc login --server=${KUBERNETES_MASTER} --certificate-authority='${MASTER_CONFIG_DIR}/ca.crt' -u new-and-unknown-user -p anything"
+os::cmd::expect_success_and_text 'oc config view' "current-context.+/${API_HOST}:${API_PORT}/new-and-unknown-user"
+# denies access after logging out
+os::cmd::expect_success 'oc logout'
```

Comment:
It's pre-existing test code so I don't think we should worry about it ... but a follow-up to determine what, exactly, it is testing could be useful. Here's sample output from a successful test run:
```
hack/test-cmd.sh:297: executing 'oc logout' expecting success
SUCCESS after 0.246s: hack/test-cmd.sh:297: executing 'oc logout' expecting success
Standard output from the command:
Logged "new-and-unknown-user" out on "https://127.0.0.1:28443"

There was no error output from the command.
```